tests/UI/campaigns/regression/checkout/100PercentDiscount_FO.js

```diff
@@ -0,0 +1,417 @@
+        this,
+        'testIdentifier',
+        'proceedToCheckout',
+        baseContext,
+      );
+
+      await cartPage.clickOnProceedToCheckout(page);
+      const isCheckoutPage = await checkoutPage.isCheckoutPage(page);
+      await expect(isCheckoutPage).to.be.true;
+    });
+
+    it('should fill personal information as a guest', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'setPersonalInformation',
+        baseContext,
+      );
+
+      const isStepPersonalInfoCompleted = await checkoutPage.setGuestPersonalInformation(
+        page,
+        customerData,
+      );
+      await expect(
+        isStepPersonalInfoCompleted,
+        'Step personal information is not completed',
+      ).to.be.true;
+    });
+
+    it('should fill address form and go to delivery step', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'setAddressStep',
+        baseContext,
+      );
+
+      const isStepAddressComplete = await checkoutPage.setAddress(
+        page,
+        addressData,
+      );
+
+      await expect(isStepAddressComplete, 'Step Address is not complete')
+        .to.be.true;
+    });
+
+    it('should go to last step', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'goToLastStep',
+        baseContext,
+      );
+
+      // Delivery step - Go to payment step
+      const isStepDeliveryComplete = await checkoutPage.goToPaymentStep(page);
+      await expect(isStepDeliveryComplete, 'Step Address is not complete')
+        .to.be.true;
+    });
+
+    it('should contain no payment needed text', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'checkNoPaymentNeededText',
+        baseContext,
+      );
+
+      const noPaymentNeededText = await checkoutPage.getTextContent(
+        page,
+        checkoutPage.noPaymentNeededElement,
+      );
+
+      await expect(noPaymentNeededText).to.contains(checkoutPage.noPaymentNeededText);
+    });
+
+    it('complete order button should be enabled', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'checkCompleteIsNotDisabled',
+        baseContext,
+      );
+
+      const confirmButtonVisible = await checkoutPage.isPaymentConfirmationButtonVisibleAndEnabled(
+        page,
+      );
+
+      await expect(confirmButtonVisible, 'Confirm button visible').to.be.true;
+    });
+
+    it('should complete the order', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'completeOrder',
+        baseContext,
+      );
+
+      // and click on the confirm button should take us to the confirmationpage
+      await checkoutPage.clickAndWaitForNavigation(
+        page,
+        checkoutPage.paymentConfirmationButton,
+      );
```

Comment:
```suggestion
      await checkoutPage.orderWithoutPaymentMethod(page);
```

We avoid as much to use selectors directly in tests files (as part of the POM architecture). May i suggest to add this function in the checkout page than use it in the test.

PS: using selectors in `goToSubMenu` function is an exception, and we should work on it.

```js
  /**
   * Order when no payment is needed
   * @param page
   * @returns {Promise<void>}
   */
  async orderWithoutPaymentMethod(page) {
    // Click on terms of services checkbox if visible
    if (await this.elementVisible(page, this.conditionToApproveLabel, 500)) {
      await Promise.all([
        this.waitForVisibleSelector(page, this.paymentConfirmationButton),
        page.click(this.conditionToApproveLabel),
      ]);
    }

    // Validate the order
    await this.clickAndWaitForNavigation(page, this.paymentConfirmationButton);
  }
  ```

Reply:
Yup, makes sense. Done in 39e0147 thanks!